coffea/processor/executor.py

```diff
@@ -99,16 +127,126 @@ def futures_executor(items, function, accumulator, workers=1, status=True, unit=
             Label of progress bar unit
         desc : str
             Label of progress bar description
+        compressed : bool, optional
+            Expect function to return compressed accumulator output (default: false)
+            This option should match the configuration of _work_function and is thus
+            somewhat internal.
     """
+    workers = kwargs.pop('workers', 1)
+    status = kwargs.pop('status', True)
+    unit = kwargs.pop('unit', 'items')
+    desc = kwargs.pop('desc', 'Processing')
+    compressed = kwargs.pop('compressed', False)
+    add_fn = _compressed_iadd if compressed else _iadd
     with concurrent.futures.ProcessPoolExecutor(max_workers=workers) as executor:
         futures = set()
-        futures.update(executor.submit(function, item, **kwargs) for item in items)
-        futures_handler(futures, accumulator, status, unit, desc)
+        futures.update(executor.submit(function, item) for item in items)
+        _futures_handler(futures, accumulator, status, unit, desc, add_fn)
+    return accumulator
+
+
+def dask_executor(items, function, accumulator, **kwargs):
+    """Execute using dask futures
+
+    Parameters
+    ----------
+        items : list
+            List of input arguments
+        function : function
+            A function to be called on each input, which returns an accumulator instance
+        accumulator : AccumulatorABC
+            An accumulator to collect the output of the function
+        client : distributed.client.Client
+            A dask distributed client instance
+        treereduction : int, optional
+            Tree reduction factor for output accumulators (default: 20)
+        status : bool, optional
+            If true (default), enable progress bar
+        compressed : bool, optional
+            Expect function to return compressed accumulator output (default: false)
+            This option should match the configuration of _work_function and is thus
+            somewhat internal.
+    """
+    client = kwargs.pop('client')
+    ntree = kwargs.pop('treereduction', 20)
+    status = kwargs.pop('status', True)
+    compressed = kwargs.pop('compressed', False)
+    add_fn = _compressed_iadd if compressed else _iadd
+    reducer = partial(_dask_reduce, add_fn=add_fn)
+
+    futures = client.map(function, items)
+    while len(futures) > 1:
+        futures = client.map(
+            reducer,
+            [futures[i * ntree:(i + 1) * ntree] for i in range(len(futures) // ntree + 1)],
+            priority=1,
+        )
+    if status:
+        from dask.distributed import progress
+        progress(futures, multi=False, notebook=False)
+    accumulator += futures[0].result()
+    return accumulator
+
+
+def parsl_executor(items, function, accumulator, **kwargs):
```

Comment:
@lgray I actually started on the parsl conversion, leaving a sketch here. (it's not forwarded to `__init__.py` yet)

Reply:
OK - cool. I'll think about spark then.